marklogic-client-api/src/test/java/com/marklogic/client/test/LineSplitterTest.java

```diff
@@ -0,0 +1,100 @@
+    private DatabaseClient client;
+
+    @Before
+    public void setUp() {
+        client = DatabaseClientFactory.newClient("localhost", 8012,
+                new DatabaseClientFactory.DigestAuthContext("rest-admin", "x"));
+    }
+
+    @Test
+    public void testSplitter() throws Exception {
+        LineSplitter splitter = new LineSplitter();
+        Stream<StringHandle> contentStream = splitter.split(new FileInputStream(jsonlFile));
+        assertNotNull(contentStream);
+
+        StringHandle[] result = contentStream.toArray(size -> new StringHandle[size]);
+        assertNotNull(result);
+        assertEquals(result.length, Files.lines(Paths.get(jsonlFile)).count());
+
+        String[] originalResult = Files.lines(Paths.get(jsonlFile))
+                                    .toArray(size -> new String[size]);
+
+        for (int i = 0; i < result.length && i < originalResult.length; i++) {
+            assertNotNull(result[i].get());
+            assertEquals(result[i].get(), originalResult[i]);
+        }
+    }
+
+    @Test
+    public void testSplitterGzip() throws Exception {
+        LineSplitter splitter = new LineSplitter();
+        GZIPInputStream gzipStream = new GZIPInputStream(new FileInputStream(jsonlGzipFile));
+        Stream<StringHandle> contentStream = splitter.split(gzipStream);
+        assertNotNull(contentStream);
+
+        StringHandle[] result = contentStream.toArray(size -> new StringHandle[size]);
+        assertNotNull(result);
+
+        gzipStream = new GZIPInputStream(new FileInputStream(jsonlGzipFile));
+        String[] originalResult = new BufferedReader(new InputStreamReader(gzipStream))
+                                    .lines()
+                                    .toArray(size -> new String[size]);
+        assertEquals(result.length, originalResult.length);
+
+        for (int i = 0; i < result.length && i < originalResult.length; i++) {
+            assertNotNull(result[i].get());
+            assertEquals(result[i].get(), originalResult[i]);
+        }
+    }
+
+    @Test
+    public void testSplitterXML() throws Exception {
+        LineSplitter splitter = new LineSplitter();
+        splitter.setFormat(Format.XML);
+        Stream<StringHandle> contentStream = splitter.split(new FileInputStream(xmlFile));
+        assertNotNull(contentStream);
+
+        StringHandle[] result = contentStream.toArray(size -> new StringHandle[size]);
+        assertNotNull(result);
+        assertEquals(result.length, Files.lines(Paths.get(xmlFile)).count());
+
+        String[] originalResult = Files.lines(Paths.get(xmlFile))
+                                    .toArray(size -> new String[size]);
+        assertEquals(result[0].getFormat(), Format.XML);
+        for (int i = 0; i < result.length && i < originalResult.length; i++) {
+            assertNotNull(result[i].get());
+            assertEquals(result[i].get(), originalResult[i]);
+        }
```

Comment:
One other thought.

All 3 tests share the same fundamental logic for inspecting the results of splitting.  Would it make sense to create a private function that takes a Stream<StringHandle>, a Format, and the String[] array of file lines and is called from each of the three tests to perform the inspection common to the three tests?

If that's possible, that would make maintenance easier (with the cost of having to break out separate tests if we ever discover a need for different inspection down the road).

Reply:
added below method
private void checkContent(Stream<StringHandle> contentStream,
                                 Format format,
                                 String[] originalResult) {...}